okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -599,6 +603,29 @@ public static HttpUrl get(URL url) {
     return parse(url.toString());
   }
 
+  /**
+   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
+   * URL, or throws an exception if it isn't.
+   *
+   * @throws MalformedURLException if there was a non-host related URL issue
+   * @throws UnknownHostException if the host was invalid
+   */
+  static HttpUrl getChecked(URL url) throws MalformedURLException, UnknownHostException {
```

Comment:
Mind making this method take a `String`? Then the caller can call `url.toString()`, which potentially makes this slightly more applicable.